internal/http/services/owncloud/ocdav/spaces.go

```diff
@@ -135,3 +144,56 @@ func (h *SpacesHandler) Handler(s *svc) http.Handler {
 		}
 	})
 }
+
+func (h *SpacesHandler) handleSpacesTrashbin(w http.ResponseWriter, r *http.Request, s *svc, trashbinHandler *TrashbinHandler) {
+	ctx := r.Context()
+	log := appctx.GetLogger(ctx)
+
+	var spaceID string
+	spaceID, r.URL.Path = router.ShiftPath(r.URL.Path)
+	if spaceID == "" {
+		// listing is disabled, no auth will change that
+		w.WriteHeader(http.StatusMethodNotAllowed)
+		return
+	}
+
+	ref := &provider.Reference{
+		ResourceId: &provider.ResourceId{
+			StorageId: spaceID,
+			OpaqueId:  spaceID,
+		},
+	}
+
+	var key string
+	key, r.URL.Path = router.ShiftPath(r.URL.Path)
+
+	switch r.Method {
+	case MethodPropfind:
+		trashbinHandler.listTrashbin(w, r, s, ref, path.Join(_trashbinPath, spaceID), key, r.URL.Path)
+	case MethodMove:
+		if key == "" {
+			http.Error(w, "501 Not implemented", http.StatusNotImplemented)
```

Comment:
Curious: Why `501`? Isn't it more like a `Bad Request`?

Reply:
Maybe but it was already there and is probably checked by the integration tests so I won't change it.